js/lib/urlutil.js

```diff
@@ -508,6 +508,14 @@ const UrlUtil = {
     return ip.isPrivate(hostname) || hostname === 'localhost' || whitelistSuffixes.some((suffix) => {
       return hostname && hostname.endsWith(`.${suffix}`)
     })
+  },
+
+  shortenUrlToDomainTLD: (url) => {
```

Comment:
can you just use `getBaseDomain` from js/lib/baseDomain.js ?

Reply:
nvm, i guess what you want here is the full URL without the path?

this seems like it would fail on URLs that don't include slashes in the protocol, like `mailto:` 

you could do `parsed = urlParse(url); return parsed.path ? parsed.split(parsed.path)[0] : url` or something like that

Reply:
^ though i guess that doesn't work with URLs that don't have a protocol

since publishers only works on http or https sites, you could fix by
```
if (!url.startsWith('http://') && !url.startsWith('https://') { url = `https://${url}` }
```

Reply:
@diracdeltas That works well. Thanks!

Reply:
@diracdeltas I'm going to go with a combo and also to make sure another protocol isn't passed in.
```if (!url.startsWith('http://') && !url.startsWith('https://') && !url.includes('://')) { url = `https://${url}` }```